sage.py

```diff
@@ -481,1274 +423,920 @@ def getepisodes(action_seq, mcat, plot, debug=False):
+                color = tcols[team_attacker.split('-')[0]]  # Team color
+                # _print_unique_attempts(team_attacker, attempts)
+
+                for attempt in attempts:
+                    # Record signatures
+                    for action in attempt:  # action = (vert_name, start_time, end_time, signs, timestamps)
+                        if action[0] not in node_signatures.keys():
+                            node_signatures[action[0]] = set()
+                        node_signatures[action[0]].update(action[3])
+
+                    # Create nodes for the AG
+                    for vid, action in enumerate(attempt):  # Iterate over each action in an attempt
+                        vname = action[0]
+                        if vid == 0:  # If first action
+                            if 'Sink' in vname:  # If sink, make dotted TODO: this check is always False
+                                lines.append('"' + _translate(vname) + '" [style="dotted,filled", fillcolor= yellow]')
                             else:
-                                sinkflag = False
-                                for sink in sev_sinks:    
-                                    if vname.split("|")[-1] == sink: # else if a high-sev sink, make dotted too
-                                        sinkflag = True
-                                        break
-                                if sinkflag:
-                                    lines.append((0,'"'+translate(vname)+'" [style="dotted,filled", fillcolor= yellow]'))
+                                if _is_severe_sink(vname, sev_sinks):  # If med- or high-sev sink, make dotted
+                                    lines.append('"' + _translate(vname) + '" [style="dotted,filled", fillcolor= yellow]')
                                     already_addressed.add(vname.split('|')[2])
-                                else: # else, normal starting node
-                                    lines.append((0,'"'+translate(vname)+'" [style=filled, fillcolor= yellow]'))
-                        else: # for other actions
-                            if 'Sink' in vname: # if sink
-                                line = [x[1] for x in lines] # take all AG graph lines so far, and see if it was ever defined before, re-define it to be dotted
-                                quit = False
-                                for l in line:
-                                    if (translate(vname) in l) and ('dotted' in l) and ('->' not in l): # if already defined as dotted, move on
-                                        quit = True
+                                else:  # Else, normal starting node
+                                    lines.append('"' + _translate(vname) + '" [style=filled, fillcolor= yellow]')
+                        else:  # For other actions
+                            if 'Sink' in vname:  # TODO: this check is always false
```

Comment:
This is not executed (see comment above). Also, in this case, the `else: # For other actions` block is not doing anything, since only here is the if-check for`Sink`

Reply:
For a better view, this is the original code on `main` branch:

![image](https://github.com/tudelft-cda-lab/SAGE/assets/45541174/9dddd826-2ec9-4dfc-8f88-6805c1ab6410)

The `if 'Sink' in vname:` part is always false, as discussed above